src/bin/main.rs

```diff
@@ -51,7 +53,18 @@ pub struct Args {
 
 fn run (args: Args) -> Result<(), Error> {
     let file = File::create(Path::new(&args.filename))?;
-    let mut obj = ArtifactBuilder::new(Target::X86_64)
+    let target = Triple {
+        architecture: Architecture::X86_64,
+        vendor: Vendor::Unknown,
+        operating_system: OperatingSystem::Unknown,
+        environment: Environment::Unknown,
+        binary_format: if args.mach {
```

Comment:
That this is a field also gives me hope that the binary format isn’t conflated/tightly coupled with the with the OS target, etc.

In particular, this is one aspect of traditional string triples that really bothers me and I hope faerie can (still) be more flexible about - eg I tell you the machine target and the binary format and don’t need to specify anything else.